### ISSUE TITLE
[no-unused-vars] argsIgnorePattern not working when unpacking

### ISSUE THREAD

**Issue by Alwinator** [typescript-eslint/typescript-eslint#4876]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```js
{
  rules: {
    '@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '^_' }]
  }
}
```

```TS
const {owner: _valueToRemove, ...valuesINeed} = myObject
```

**Expected Result**
No error. It should work without a problem because the owner I want to remove is renamed to _valueToRemove. It starts with a _ so there should be no error.

**Actual Result**
ESLint: '_valueToRemove' is assigned a value but never used.(@typescript-eslint/no-unused-vars)

Also _valueToRemove, is underlined red in my IDE (WebStorms)

**Additional Info**
tsconfig
```JSON
{
  "compilerOptions": {
    "module": "commonjs",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "target": "esnext",
    "sourceMap": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "incremental": true,
    "skipLibCheck": true,
    "strictNullChecks": false,
    "noImplicitAny": false,
    "strictBindCallApply": false,
    "forceConsistentCasingInFileNames": false,
    "noFallthroughCasesInSwitch": false,
    "resolveJsonModule": true,
  }
}
```

Did eslint throw an exception?
```
eslint test-suite/tests/mytest.e2e-spec.ts 

/home/alwin/WebStormsProjects/myproject/test-suite/tests/mytest.e2e-spec.ts
  79:19  error  '_useless' is assigned a value but never used  @typescript-eslint/no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `canary` |
| `@typescript-eslint/parser`        | `canary` |
| `TypeScript`                       | `4.6.3` |
| `ESLint`                           | `8.13.0` |
| `node`                             | `8.13.0` |

**Comment by bradzacher** [comment]
It's not supposed to.
As it's name suggest it only applies to ***arguments***.

You can ignore rest siblings with the option `ignoreRestSiblings`
https://eslint.org/docs/rules/no-unused-vars#ignorerestsiblings

**Comment by Alwinator** [comment]
@bradzacher Okay, I didn't know that! Sorry for the wrong report.
